box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsDataWrapper.java

```diff
@@ -0,0 +1,40 @@
+package com.box.androidsdk.share.vm;
+
+/**
+ * A special extension of DataWrapper to hold data for inviting a new Collaborator
+ */
+public class InviteCollaboratorsDataWrapper extends DataWrapper<String> {
+
+    private boolean mInvitationFailed;
+
+    public InviteCollaboratorsDataWrapper(String data, int strCode, boolean invitationFailed) {
+        this.mData = data;
+        this.mStrRes = strCode;
+        this.mInvitationFailed = invitationFailed;
+    }
+
+    /**
+     * Returns true if all invitations succeeded.
+     * @return true if all invitations succeeded
+     */
+    @Override
+    public boolean isSuccess() {
+        return !isInvitationFailed();
+    }
+
+    /**
+     * Returns true if at least one invitation failed.
+     * @return true if at least one invitation failed
+     */
+    public boolean isInvitationFailed() {
+        return mInvitationFailed;
+    }
+
+    /**
+     * Returns true if the String formatted part exists.
+     * @return true if the String formatted part exists
+     */
+    public boolean hasSubMessage() {
+        return mData != null;
```

Comment:
looks like super.isSuccess()

Reply:
That's true. But I feel that's too confusing since this child class has its old definition of isSuccess() already. having a subMessage() is not necessarily a successful request either.